ambry-tools/src/test/java/com/github/ambry/clustermap/HelixBootstrapUpgradeToolTest.java

```diff
@@ -0,0 +1,399 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.clustermap;
+
+import com.github.ambry.clustermap.TestUtils.TestHardwareLayout;
+import com.github.ambry.clustermap.TestUtils.TestPartitionLayout;
+import com.github.ambry.utils.Utils;
+import com.google.common.collect.Sets;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.I0Itec.zkclient.IDefaultNameSpace;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.ZkServer;
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.model.IdealState;
+import org.apache.helix.model.InstanceConfig;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static com.github.ambry.clustermap.HelixBootstrapUpgradeTool.*;
+
+
+public class HelixBootstrapUpgradeToolTest {
+  /**
+   * A class to initialize and hold information about each Zk Server.
+   */
+  private static class ZkInfo {
+    String dcName;
+    int port;
+    String dataDir;
+    String logDir;
+    ZkServer zkServer;
+
+    /**
+     * Instantiate by starting a Zk server.
+     * @param dcName the name of the datacenter.
+     * @param port the port at which this Zk server should run on localhost.
+     */
+    ZkInfo(String dcName, int port) throws IOException {
+      this.dcName = dcName;
+      this.port = port;
+      File tempDir = Files.createTempDirectory(dcName).toFile();
+      tempDirPath = tempDir.getAbsolutePath();
+      tempDir.deleteOnExit();
+      this.dataDir = tempDirPath + "/dataDir";
+      this.logDir = tempDirPath + "/logDir";
+      startZkServer(port, dataDir, logDir);
+    }
+
+    private void startZkServer(int port, String dataDir, String logDir) {
+      IDefaultNameSpace defaultNameSpace = new IDefaultNameSpace() {
+        @Override
+        public void createDefaultNameSpace(ZkClient zkClient) {
+        }
+      };
+      // start zookeeper
+      zkServer = new ZkServer(dataDir, logDir, defaultNameSpace, port);
+      zkServer.start();
+    }
+
+    void shutdown() {
+      zkServer.shutdown();
```

Comment:
they get deleted because a `deleteOnExit()` call is made on the root directory after creation.